drivers/hmis_external_apis/extensions/hmis/hud/client_extension.rb

```diff
@@ -42,6 +48,29 @@ def self.search_by_external_id(where, text)
 
             where.or(arel_table[:id].in(client_ids))
           end
+
+          private def validate_mci_id_exists
+            return unless HmisExternalApis::AcHmis::Mci.enabled?
+            # Valid if client has an MCI ID, or is going to create one
+            return if ac_hmis_mci_ids.exists? || send(:create_mci_id)
+
+            # If brand new Client record (NOT in context of an Enrollment), then MCI clearance is required
+            clearance_required = true if new_record?
+
+            if persisted?
+              # MCI clearance is required unless this client is ONLY enrolled at SO/NBN program(s)
+              ptypes = HmisExternalApis::AcHmis::Mci::PROJECT_TYPES_NOT_REQUIRING_CLEARANCE
+              so_nbn_enrollment_count = enrollments.with_project_type(ptypes).size
```

Comment:
This probably makes sense, but I'd so much rather defer the required-ness to the project.  That way if we decide to set that as a project attribute, so we can override the defaults, it would be much easier.

Reply:
Interesting thought. We could have a custom data element on Project for "mci clearance required." However I think at this point, thats just more custom data elements to maintain. It doesn't sound like they need to toggle this at the individual project level.

I do worry, though, that if/when the MCI API goes down or is unreachable, providers at most project types will be completely unable to enroll new clients, or edit existing client records. The nice thing about making it configurable would be that we can turn off the requirement in such a situation.

Reply:
That does seem like a reasonable reason.  I was actually thinking for now you'd call `project.mci_clearance_required?` and it would just be something like:
``` 
def .mci_clearance_required?
 project_type.in?(HmisExternalApis::AcHmis::Mci::PROJECT_TYPES_NOT_REQUIRING_CLEARANCE)
end
```

We could come back later and upgrade it to a field, or some other calculation.

Reply:
Oh I see. Yeah.